### PR TITLE
Restore time sig as well in restoreInitialKeySig after deleting all measures then appending one

### PR DESCRIPTION
**Pull request by Nekotizimo**
Resolves: #17336

After deleting all measures then appending a measure, the key signature is reset to the default C major, but there is no time signature. It now adds a 4/4 time signature, which I think is the default.

Perhaps it should restore the previously used time and key signatures, instead of reverting to C major and 4/4. Personally I think it should reset to default, since if you want to preserve the time and key signatures, you would only delete the notes (Delete) instead of deleting the measures (Cmd-Delete).

Let me know if there are any issues.

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [x] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [x] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

**Review by oktophonie — APPROVED** [review]

File: src/engraving/libmscore/edit.cpp [hunk 1]
```diff
@@ -4264,6 +4265,22 @@ void Score::restoreInitialKeySig()
...
+        undoAddElement(newTimeSig);
+
+        SigEvent timeSigEvent;
+        timeSigEvent.setTimesig(Fraction(4, 4));
+        timeSigEvent.setNominal(Fraction(4, 4));
+
+        firstMeas->setTimesig(newTimeSig->sig());
+        firstMeas->setTicks(newTimeSig->sig());
+        staff->addTimeSig(newTimeSig);
+        sigmap()->add(0, timeSigEvent);
```
Comment (cbjeukendrup) [inline src/engraving/libmscore/edit.cpp:4283]: `firstMeas->setTimesig`, `firstMeas->setTicks` and `sigmap()->add` only need to be called once, instead of for every staff again. So it would be good to take them out of the for-loop. Also, I'm not sure if calling `staff->addTimeSig` and `sigmap()->add` is really necessary; it looks like that will be done in `Score::rebuildTempoAndTimeSigMaps`, which will eventually be called anyway after `Score::insertMeasure`.

**Comment by Nekotizimo** [comment]
Thanks! I'm also looking into the failed `splitTests`, so I'll update this with another commit.

File: src/engraving/tests/split_data/split01.mscx [hunk 2]
```diff
@@ -1,38 +1,40 @@
...
     <metaTag name="workTitle">Test</metaTag>
-    <Part>
+    <Part id="1">
       <Staff id="1">
         <StaffType group="pitched">
           <name>stdNormal</name>
           </StaffType>
         </Staff>
       <trackName>Voice</trackName>
-      <Instrument>
```
Comment (cbjeukendrup) [inline src/engraving/tests/split_data/split01.mscx:30]: What is the reason for this change? (and the similar ones)

**Comment by cbjeukendrup** [comment]
Also, a rebase is needed (albeit a trivial one; GitHub is just upset because some files were moved)

File: src/engraving/tests/split_data/split01.mscx (hunk 2 above)
Comment (Nekotizimo) [reply src/engraving/tests/split_data/split01.mscx:30]: These tests should be updated since it currently incorrectly adds a C Major key signature to the second measure after splitting the measure when it shouldn't modify the key signature. 

Frankly, I'm not sure how to make the reference files properly (without the unnecessary metadata etc), so I just remade both the before and after (ref) files again, which resulted in the different instruments, part ids, etc. Let me know if I should do something else instead.

Comment (cbjeukendrup) [reply src/engraving/tests/split_data/split01.mscx:30]: The core of the problem is actually that these test files follow an old version of the file format, because they were never updated; and, apart from that just nobody has made the effort, the reason that they were never updated, is that saving mscx files (rather than mscz, which are binary files and thus not usable for comparisons) has become a lot more cumbersome in MS4. 

So, usually we just look if the differences between the ref xml and the result of the test make sense, and then edit the ref file accordingly... 

Anyway, the changes mostly seem fine, I've only a few requests:
- It would make more sense to change the instrument IDs to voice, actually like it was in the ref files
- There are some undesired whitespace changes in the non-ref files, like spaces added/removed at the beginning/end of the line. Would be great if you could fix those.

Comment (Nekotizimo) [reply src/engraving/tests/split_data/split01.mscx]: Makes sense. Didn't realize all the awful whitespaces I added to the files, that's my bad.
I've reverted the tests back to what's in master currently, and only removed the key sigs from the ref xml.

**Review by cbjeukendrup — APPROVED** [review]
